### ISSUE TITLE
Nvidia GPU driver addons init container fails

### ISSUE THREAD

**Issue by Zubnix** [kubernetes/minikube#15123]
### What Happened?

Looks like it requires python3

init container log output:
```
  HOSTCC  scripts/selinux/mdp/mdp
  HOSTCC  scripts/kallsyms
  HOSTCC  scripts/recordmcount
  HOSTCC  scripts/sorttable
  HOSTCC  scripts/asn1_compiler
  HOSTCC  scripts/extract-cert
  WRAP    arch/x86/include/generated/uapi/asm/bpf_perf_event.h
  WRAP    arch/x86/include/generated/uapi/asm/errno.h
  WRAP    arch/x86/include/generated/uapi/asm/fcntl.h
  WRAP    arch/x86/include/generated/uapi/asm/ioctl.h
  WRAP    arch/x86/include/generated/uapi/asm/ioctls.h
  WRAP    arch/x86/include/generated/uapi/asm/ipcbuf.h
  WRAP    arch/x86/include/generated/uapi/asm/param.h
  WRAP    arch/x86/include/generated/uapi/asm/poll.h
  WRAP    arch/x86/include/generated/uapi/asm/resource.h
  WRAP    arch/x86/include/generated/uapi/asm/socket.h
  WRAP    arch/x86/include/generated/uapi/asm/sockios.h
  WRAP    arch/x86/include/generated/uapi/asm/termbits.h
  WRAP    arch/x86/include/generated/uapi/asm/termios.h
  WRAP    arch/x86/include/generated/uapi/asm/types.h
  WRAP    arch/x86/include/generated/asm/early_ioremap.h
  WRAP    arch/x86/include/generated/asm/export.h
  WRAP    arch/x86/include/generated/asm/mcs_spinlock.h
  WRAP    arch/x86/include/generated/asm/irq_regs.h
  WRAP    arch/x86/include/generated/asm/local64.h
  WRAP    arch/x86/include/generated/asm/mm-arch-hooks.h
  WRAP    arch/x86/include/generated/asm/mmiowb.h
  WRAP    arch/x86/include/generated/asm/module.lds.h
  WRAP    arch/x86/include/generated/asm/rwonce.h
  UPD     include/config/kernel.release
  UPD     include/generated/uapi/linux/version.h
  UPD     include/generated/utsrelease.h
  CC      scripts/mod/empty.o
  HOSTCC  scripts/mod/mk_elfconfig
  MKELF   scripts/mod/elfconfig.h
  HOSTCC  scripts/mod/modpost.o
  CC      scripts/mod/devicetable-offsets.s
  UPD     scripts/mod/devicetable-offsets.h
  HOSTCC  scripts/mod/file2alias.o
  HOSTCC  scripts/mod/sumversion.o
  HOSTLD  scripts/mod/modpost
  CC      kernel/bounds.s
  UPD     include/generated/bounds.h
  UPD     include/generated/timeconst.h
  CC      arch/x86/kernel/asm-offsets.s
  UPD     include/generated/asm-offsets.h
  CALL    scripts/checksyscalls.sh
  CALL    scripts/atomic/check-atomics.sh
  DESCEND  objtool
  HOSTCC   /usr/src/linux-5.10.57/tools/objtool/fixdep.o
  HOSTLD   /usr/src/linux-5.10.57/tools/objtool/fixdep-in.o
  LINK     /usr/src/linux-5.10.57/tools/objtool/fixdep
  CC       /usr/src/linux-5.10.57/tools/objtool/exec-cmd.o
  CC       /usr/src/linux-5.10.57/tools/objtool/help.o
  CC       /usr/src/linux-5.10.57/tools/objtool/pager.o
  CC       /usr/src/linux-5.10.57/tools/objtool/parse-options.o
  CC       /usr/src/linux-5.10.57/tools/objtool/run-command.o
  CC       /usr/src/linux-5.10.57/tools/objtool/sigchain.o
  CC       /usr/src/linux-5.10.57/tools/objtool/subcmd-config.o
  LD       /usr/src/linux-5.10.57/tools/objtool/libsubcmd-in.o
  AR       /usr/src/linux-5.10.57/tools/objtool/libsubcmd.a
  CC       /usr/src/linux-5.10.57/tools/objtool/arch/x86/special.o
  MKDIR    /usr/src/linux-5.10.57/tools/objtool/arch/x86/lib/
  GEN      /usr/src/linux-5.10.57/tools/objtool/arch/x86/lib/inat-tables.c
  CC       /usr/src/linux-5.10.57/tools/objtool/arch/x86/decode.o
  LD       /usr/src/linux-5.10.57/tools/objtool/arch/x86/objtool-in.o
  CC       /usr/src/linux-5.10.57/tools/objtool/weak.o
  CC       /usr/src/linux-5.10.57/tools/objtool/check.o
  CC       /usr/src/linux-5.10.57/tools/objtool/special.o
  CC       /usr/src/linux-5.10.57/tools/objtool/orc_gen.o
  CC       /usr/src/linux-5.10.57/tools/objtool/orc_dump.o
  CC       /usr/src/linux-5.10.57/tools/objtool/builtin-check.o
  CC       /usr/src/linux-5.10.57/tools/objtool/builtin-orc.o
  CC       /usr/src/linux-5.10.57/tools/objtool/elf.o
  CC       /usr/src/linux-5.10.57/tools/objtool/objtool.o
  CC       /usr/src/linux-5.10.57/tools/objtool/libstring.o
  CC       /usr/src/linux-5.10.57/tools/objtool/libctype.o
  CC       /usr/src/linux-5.10.57/tools/objtool/str_error_r.o
  CC       /usr/src/linux-5.10.57/tools/objtool/librbtree.o
  LD       /usr/src/linux-5.10.57/tools/objtool/objtool-in.o
  LINK     /usr/src/linux-5.10.57/tools/objtool/objtool
  DESCEND  bpf/resolve_btfids
  MKDIR     /usr/src/linux-5.10.57/tools/bpf/resolve_btfids//libbpf
Auto-detecting system features:
...                        libelf: [ on  ]
...                          zlib: [ on  ]
...                           bpf: [ on  ]
  GEN      /usr/src/linux-5.10.57/tools/bpf/resolve_btfids/libbpf/bpf_helper_defs.h
/usr/bin/env: 'python3': No such file or directory
Makefile:182: recipe for target '/usr/src/linux-5.10.57/tools/bpf/resolve_btfids/libbpf/bpf_helper_defs.h' failed
make[3]: *** [/usr/src/linux-5.10.57/tools/bpf/resolve_btfids/libbpf/bpf_helper_defs.h] Error 127
make[3]: *** Deleting file '/usr/src/linux-5.10.57/tools/bpf/resolve_btfids/libbpf/bpf_helper_defs.h'
Makefile:44: recipe for target '/usr/src/linux-5.10.57/tools/bpf/resolve_btfids//libbpf/libbpf.a' failed
make[2]: *** [/usr/src/linux-5.10.57/tools/bpf/resolve_btfids//libbpf/libbpf.a] Error 2
Makefile:71: recipe for target 'bpf/resolve_btfids' failed
make[1]: *** [bpf/resolve_btfids] Error 2
Makefile:1947: recipe for target 'tools/bpf/resolve_btfids' failed
make: *** [tools/bpf/resolve_btfids] Error 2
```

Minikube was setup using the official kvm2+gpu passthrough instructions and using command:
`minikube start --addons=ingress --addons=ingress-dns --addons=nvidia-gpu-device-plugin --addons=nvidia-driver-installer --driver=kvm2 --kvm-gpu --container-runtime=containerd --kubernetes-version=1.24.6`

### Attach the log file

[log.txt](https://github.com/kubernetes/minikube/files/9762897/log.txt)

### Operating System

Ubuntu

### Driver

KVM2

**Comment by Zubnix** [comment]
After manually adjusting the driver image so it has python3, the image correctly installs the driver, however it fails at the end, claiming no cards are found. On further inspection, the nvidia drivers are build & loaded by the kernel.

Furthermore it looks like container runtime (containerd in this case) is not correctly adjusted to utilize the GPU so no pods get to actually use the GPU. Using the official nvidia helm chart: https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/getting-started.html#bare-metal-passthrough-with-pre-installed-nvidia-drivers which should take care of this fails because nvidia-smi is not installed by the drivers.

In short it looks like the officially supported kvm2 GPU passthrough method is completely broken for the moment.

**Comment by joysn71** [comment]
I ran into the same issue. How did you fix the image?

**Comment by Zubnix** [comment]
By using something else than minikube

**Comment by joysn71** [comment]
Meanwhile I was able to fi the missing Python issue, but now nvidia-smi reports that no GPU is found.
Which replacement did you use, if I may ask? :)

**Comment by dulacp** [comment]
I got it working just like @Zubnix by adding the python3 dependency, as you can see in my PR to [GoogleCloudPlatform/container-engine-accelerators](https://github.com/GoogleCloudPlatform/container-engine-accelerators/pull/350).

Then build the Docker image and update the minikube addon to use this image instead of the default one:

```console
# Build the new image
eval $(minikube docker-env)
TAG=$(git rev-parse HEAD)
docker build --pull -t minikube-nvidia-driver-installer:$TAG .

# Update minikube addon
minikube addons enable nvidia-driver-installer \
    --images="NvidiaDriverInstaller=minikube-nvidia-driver-installer:$TAG" \
    --registries="NvidiaDriverInstaller="
```
